src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+
+    try:
+        if storage_space_available :
+
+            # Creating new Deployment Logs folder in the given timestamp folder
+            deployments_path = os.path.join(filepath_with_timestamp, "Deployment_logs")
+            try:
+                os.mkdir(deployments_path)
+            except FileExistsError:
+                pass
+
+            # To retrieve all the the deployements that are present in the Cluster
+            deployments_list = appv1_api_instance.list_namespaced_deployment("azure-arc")
+
+            # Traversing through all the deployments present
+            for deployment in deployments_list.items:
+
+                # Fetching the deployment name
+                deployment_name = deployment.metadata.name
+
+                deployment_logs_path = os.path.join(deployments_path, deployment_name + ".txt")
+                # Creating a text file with the name of the deployment and adding deployment status in it
+                with open(deployment_logs_path, 'w+') as deployment_file:
+                    deployment_file.write(str(deployment.status))
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            
+    except Exception as e:
+        logger.warning("An expection has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+    
+    return storage_space_available
+
+
+def check_agent_state(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    try:
+       
+        counter = 0
+        agent_state_path=os.path.join(filepath_with_timestamp, "Agent_State.txt")
+        with open(agent_state_path, 'w+') as agent_state:
+            # To retrieve all of the arc agent pods that are presesnt in the Cluster
+            arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+            # Check if any arc agent other than kube aadp proxy is not in Running state
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                if storage_space_available :
+
+
+                    # Storing the state of the arc agent in the user machine
+                    agent_state.write(each_agent_pod.metadata.name + " = " + each_agent_pod.status.phase +"\n")
+
+                # If the agent is Kube add proxy we will continue with the next agent
+                if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                    continue
+                if each_agent_pod.status.phase != 'Running':
+                    counter = 1
+
+        # Displaying error if the arc agents are in pending state.
+        if counter:
+            print("Error: One or more Azure Arc agents are in pending state. It may be caused due to insufficient resource availability on the cluster.\n For more details on resource requirement visit 'aka.ms\\arcenabledkubernetesresourcerequirement'. \n")
```

Comment:
Since we don't have the link for resource requirements from the PMs yet, let's remove the link part from log for now and add it as a To-do in the comment next to it. We can add it right after we get the aka.ms link